src/game.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const STRIKE = 10,    SPARE = 10,   ALL_PINS = 10,    MAX_ROLLS = 3,
+      MAX_FRAMES = 10;
+var   roll1 = 0,      roll2 = 1;
+
+
+var Game = function() {
+  this.frameCount = 0;
+  this.frames = [];
+  this.roll;
+  this.remainingPins;
+};
+
+Game.prototype = {
+
+  addBonuses() {
+    var [frames, l] = [this.frames, this.frames.length];
+    var [antepenult, penult, ult] = [frames[l - 3], frames[l - 2], frames[l -1]];
+    if (l > 2 && this.isStrike(antepenult) && !this.hasBonus(antepenult)) {
+      antepenult.push(ult[roll1]);
+      ult[roll2] ? penult.push(ult[roll1], ult[roll2]) : penult.push(ult[roll1]);
+    } else if (l > 1 && this.isStrike(penult) && !this.hasBonus(penult)) {
+      ult[roll2] && ult[roll2] !== 10 ?
```

Comment:
Whenever we see multi-pronged if statements, we should ask ourselves whether we can refactor/simplify. If we see several of these in the same class (as we do here), then that is an even stronger indication, and it usually means that we should push some logic out into another class. I mentioned your commented out frame class above - was that the original intention here?

A technique I use here is something like the 5 whys - I keep asking myself why I need the conditions. This can help me to work out whether there's a more fundamental and potentially universal truth. E.g.

Why do I have 3 conditions? 
Because the last 3 frames behave differently with regards to bonuses

Why do they behave differently with regards to bonuses?
Because strikes and half strikes need bonuses but their may not be enough frames to play

Why may there not be enough frames to play?
Because we've reached the end of the game

Why does reached the end of the game mean we don't have enough frames?
Because we want to continue rolling but we need a frame to record the score against

Why do we need a frame to record the score against?
... erm, we don't. 

At this point, I might start thinking that actually, all the frames as the same, and they all behave the same with regards to bonuses, and the only difference is that the game allows me to take a couple of extra rolls if a bonus is still pending near the end. That ends up being a much simpler case - in simplified pseudocode, it's basically something like:

```
function isGameOver() {
  return frames.find((frame) => frame.needsRolls() || frame.needsBonuses()) === null;
}

while (!isGameOver()) {
  nextRoll();
}

```